src/screens/ENSConfirmRegisterSheet.js

```diff
@@ -28,6 +47,75 @@ export default function ENSConfirmRegisterSheet() {
     rentPrice: registrationData?.rentPrice,
   });
 
+  useEffect(() => {
+    const callbackGetRentPrice = async () => {
+      const rentPrice = await getRentPrice(name, secsInYear);
+      setRentPrice(addBuffer(rentPrice.toString(), 1.1));
+    };
+    callbackGetRentPrice();
+  }, [name]);
+
+  const updateGasLimit = useCallback(async () => {
+    const salt = generateSalt();
+    const gasLimit = await getRapEstimationByType(RapActionTypes.commitENS, {
+      ensRegistrationParameters: {
+        duration: secsInYear,
+        name: name,
+        ownerAddress: accountAddress,
+        records,
+        rentPrice,
+        salt,
+      },
+    });
+    updateTxFee(gasLimit);
+    setGasLimit(gasLimit);
+  }, [accountAddress, name, records, rentPrice, updateTxFee]);
+
+  // Update gas limit
+  useEffect(() => {
+    if (!gasLimit && !isEmpty(gasFeeParamsBySpeed)) {
+      updateGasLimit();
+    }
+  }, [gasFeeParamsBySpeed, gasLimit, updateGasLimit, updateTxFee]);
+
+  useEffect(() => startPollingGasFees(), [startPollingGasFees]);
+
+  const handleCommitSubmit = useCallback(async () => {
+    const wallet = await loadWallet();
+    if (!wallet) {
+      return;
+    }
+
+    const callback = () => null;
+
+    const nonce = await getNextNonce();
+    const salt = generateSalt();
+
+    const ensRegistrationParameters = {
+      duration: secsInYear,
+      name,
+      nonce,
+      ownerAddress: accountAddress,
+      records,
+      rentPrice,
+      salt,
+    };
+
+    await dispatch(
+      saveCommitRegistrationParameters(
+        accountAddress,
+        ensRegistrationParameters
+      )
+    );
+
+    await executeRap(
+      wallet,
+      RapActionTypes.commitENS,
+      { ensRegistrationParameters },
+      callback
+    );
+  }, [accountAddress, dispatch, getNextNonce, name, records, rentPrice]);
```

Comment:
Wonder if we should hide this logic under a hook just to improve readability? good or bad idea?

```tsx
const { commit } = useENSRegistrationTransaction()
```

Reply:
i was thinking it's a good idea but not sure if it would be ideal since we're handling different tx types and we need to set different gas limits accordingly. For example, this i s getting the commit gas limit estimation on mount with the effect and the callback is in charge to execute the tx, also a callback to behave according to the execution state (success or error), feels like too many things need to be passed to the hook in order for it to work?